### DISCUSSION TITLE
Error building App in production.

### DISCUSSION THREAD

**Discussion by pragmaticAweds** [vitest-dev/vitest#6271] [Q&A]
### Describe the bug

My build fails in production, I am using vitejs, vitest, Typescript and @testing-library/jest-dom. Everything was working fine before, but the moment I added test, and push to production, it fails requesting for vite, vite/global, vitest, and  @testing-library/jest-dom. When I am in development everything is working fine, the problem only happen in production.

### Reproduction

Link to repo:  https://github.com/pragmaticAweds/test

<img width="1225" alt="Screenshot 2024-08-02 at 12 41 04 PM" src="https://github.com/user-attachments/assets/8bc580ff-5419-44bb-9aa2-c2926b8572f0">

### System Info

```shell
--npmPackages '{vitest,@vitest/*,vite,@vitejs/*}'
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by osher-scopely** [comment]
for anyone landing here years after, specifying build mode fixed it for me (`npm run build` -> `npm run build -- --mode prod`)
